packages/components/src/text-control/style.scss

```diff
@@ -14,6 +14,7 @@
 .components-text-control__input[type="number"] {
 	width: 100%;
 	height: $grid-unit-40;
+	margin: 0;
```

Comment:
Could use a comment as to why it's necessary.

Reply:
Updated by [c2bdeb5](https://github.com/WordPress/gutenberg/pull/60282/commits/c2bdeb5f2598ccfd715ddc2dc6fcaf57d6e6dce5)